### ISSUE TITLE
Docker v4.11.0 uses all RAM after a day

### ISSUE THREAD

**Issue by Zechiax** [docker/for-win#12877]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 4FDB3ACA-6B2A-4E97-A5FE-747DF9337721/20220803102008

### Actual behavior
After a ~day, Docker uses the whole RAM going from 50% usage to 97% and lags the whole computer

Although it does not appear in Task manager as Docker Desktop using the RAM, after I close it, it goes down to 30% (checked 4 times before reporting)

### Expected behavior
Using constantly the same amount of RAM

### Information

  - Windows Version: Windows 10 Pro v21H1
  - Docker Desktop Version: 4.11.0 (83626)
  - WSL2 or Hyper-V backend? WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No
  - The problem is new, it started with the 4.11.0 update

### Steps to reproduce the behavior

  1. Start docker desktop, let it run for a ~day

**Comment by blagae** [comment]
I won't claim that this will fix it, but I have sometimes found success with CPU or RAM resource issues by rebooting the WSL backend with `wsl --shutdown`.

**Comment by mrgreywater** [comment]
Can you check (using RamMap) if this is the same issue I've opened in the wsl repo?

https://github.com/microsoft/WSL/issues/8703#issuecomment-1208731450

**Comment by Zechiax** [comment]
@mrgreywater 
> Can you check (using RamMap) if this is the same issue I've opened in the wsl repo?
> 
> [microsoft/WSL#8703 (comment)](https://github.com/microsoft/WSL/issues/8703#issuecomment-1208731450)

It certainly looks like it's the same issue, after letting Docker run for about 2 hours, in RamMap there are thousands of "wsl.exe" processes and counting. 

After running `wsl --shutdown`, all the processes are still visible in RamMap.

I also looked at ProcessHacker and it's doing the same thing, creating and closing wsl.exe instance every few seconds as you have stated in your issue.


@blagae 
> I won't claim that this will fix it, but I have sometimes found success with CPU or RAM resource issues by rebooting the WSL backend with `wsl --shutdown`.

Thank you for the suggestion, it sadly doesn't work, but atleast it frees up some RAM for computer to be usable again to turn off Docker and free the whole RAM

**Comment by william00179** [comment]
I am also seeing this issue. 

Inspecting the status of the containers I have running, they are only reporting to consume ~2GB, but the RAM reported to be used in Docker desktop (the small stats on the main window) 8.9GB.

I am having to restart docker multiple times a day to free memory. 

Only started happening after recent update to 4.11.1

**Comment by MalikKillian** [comment]
Seeing same behavior as well on 4.11.1. I only identified Docker as the culprit recently but I feel like this happened before 4.11.0. I've had instability issues for a while and I always simply restarted to fix the issue. It could've been Docker the whole time.

I'm going to switch back to using Hyper-V until this is addressed.

**Comment by PassiveLemon** [comment]
I've got this issue too. I don't even have a dedicated machine for docker so this really hurts everything. I thought it was a different program so I reinstalled windows and just learned now that it is docker. Docker itself reports about 2 gigs being used (I limited docker to 5 gigs in my .wslconfig) but ends up using a little over 7. I'm on W10 using WSL2. Closing docker entirely frees all of that ram.

**Comment by aguedeney** [comment]
I had to stop using Docker 4.11.1 on Windows because of memory leakage.

**Comment by ajnicolay** [comment]
Same issue for me on 4.11.1. I have to quit docker desktop and run _wsl --shutdown_ to free up my ram. Quitting docker only returns the 2 gigs that I've allotted for it. I've had to do this at least once a day since updating to 4.11.1. RamMap shows hundreds of wsl.exe processes.

**Comment by nicks** [comment]
Thanks all for reporting! We can repro this and are investigating...

**Comment by nicks** [comment]
Thanks to everyone for the detailed repro cases; we'll have a fix for this in the next release :hugs:

**Comment by jagradang** [comment]
Do we know when the next release that fixes this will be?  this bug is causing all kinds of issues on my network currently!

**Comment by mrgreywater** [comment]
Docker 4.12 was released with no mention for a fix for this issue. I'll try and report back if it fixes this issue...

https://docs.docker.com/desktop/release-notes/#docker-desktop-4120

**Comment by ajnicolay** [comment]
> Docker 4.12 was released with no mention for a fix for this issue. I'll try and report back if it fixes this issue...
> 
> https://docs.docker.com/desktop/release-notes/#docker-desktop-4120

I noticed the same thing, no mention of a bug fix related to this. I updated to 4.12 about 18 hours ago and so far I only see 3 instances of wsl.exe in RAMMap vs 1000+ I was seeing with 4.11.1

I'll keep an eye on it.

**Comment by loucas781** [comment]
Been running the lastest version 4.12 for a couple of days now and so far my RAM usage has been pretty stable, I'd say it's been fixed with this version

**Comment by mrgreywater** [comment]
Seems to be fixed with Docker for Windows v4.12 for me aswell.

**Comment by nicks** [comment]
woooo :1st_place_medal: thank you for letting us know.

**Comment by hernandp** [comment]
Upgraded to 4.12.  No memory/CPU problem yet.

**Comment by fnils** [comment]
I have version 4.12 and today it used up all of the available memory. The only thing I know I might have done differently was that I set the computer to sleep when I went home.

**Comment by deathlyrage** [comment]
Still having this issue on 4.12

**Comment by nicks** [comment]
Sad to hear that some people are experiencing other RAM issues. 

The reason we were able to fix this issue is because the reporters posted diagnostics and detailed symptoms (which we really appreciated!). 

I've fixed enough perf issues on windows to know that they could have tons and tons of causes unrelated to Docker: a buggy anti-virus program, a misconfigured BIOS, etc, etc. If you're still experiencing RAM problems, we'll need you to file a new issue with diagnostics, or we really have no hope of being able to fix it. Thanks!
